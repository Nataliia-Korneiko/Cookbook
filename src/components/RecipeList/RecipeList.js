/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import * as recipesActions from '../../redux/recipes/recipesActions';
import RecipeItem from '../RecipeItem/RecipeItem';
import s from './RecipeList.module.css';

// const RecipeList = ({ items, onDeleteRecipe, onEditRecipe }) =>
const RecipeList = ({ items }) =>
  items.length > 0 && (
    <ul className={s.list}>
      {items.map(item => (
        <li className={s.item} key={item.id}>
          <RecipeItem
            id={item.id}
            // {...item}
            // onDeleteRecipe={() => onDeleteRecipe(item.id)}
            // onEditRecipe={() => onEditRecipe(item.id)}
          />
        </li>
      ))}
    </ul>
  );

RecipeList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
  onDeleteRecipe: PropTypes.func.isRequired,
};

// const filterRecipes = (recipes, filter) => {
//   return recipes.filter(recipe =>
//     recipe.text.toLowerCase().includes(filter.toLowerCase()),
//   );
// };

// REPETA
// const mapStateToProps = state => {
//   const normalizedFilter = state.recipes.filter.toLowerCase();

//   const visibleRecipes = state.recipes.data.filter(recipe =>
//     recipe.text.toLowerCase().includes(normalizedFilter),
//   );

//   return {
//     items: visibleRecipes,
//   };
// };

const mapStateToProps = state => ({
  items: state.recipes.data.filter(recipe =>
    recipe.text.toLowerCase().includes(state.recipes.filter.toLowerCase()),
  ),
});

// const mapDispatchToProps = dispatch => ({
//   onDeleteRecipe: id => dispatch(recipesActions.deleteRecipe(id)),
//   onEditRecipe: id => dispatch(recipesActions.updateRecipe(id)),
// });

// export default connect(mapStateToProps, mapDispatchToProps)(RecipeList);
export default connect(mapStateToProps)(RecipeList);
