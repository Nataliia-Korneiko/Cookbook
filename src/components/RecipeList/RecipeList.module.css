.list {
  display: flex;
  flex-wrap: wrap;
  justify-content: end;
  height: calc(100vh - 230px);
  overflow-y: scroll;
}

.item:not(:nth-child(4n)) {
  margin-right: 20px;
}

.item:not(:nth-last-child(-n + 4)) {
  margin-bottom: 20px;
}

@media (max-width: 1300px) {
  .item:not(:nth-child(4n)) {
    margin-right: 0;
  }

  .item:not(:nth-child(3n)) {
    margin-right: 20px;
  }

  .item:not(:nth-last-child(-n + 3)) {
    margin-bottom: 20px;
  }
}

@media (max-width: 980px) {
  .item:not(:nth-child(3n)) {
    margin-right: 0;
  }

  .item:not(:nth-child(2n)) {
    margin-right: 20px;
  }

  .item:not(:nth-last-child(-n + 2)) {
    margin-bottom: 40px;
  }
}

@media (max-width: 660px) {
  .list {
    height: calc(100vh - 246px);
  }

  .item:not(:nth-child(2n)) {
    margin-right: 0;
  }

  .item:not(:nth-last-child(-n + 1)) {
    margin-bottom: 40px;
  }
}
