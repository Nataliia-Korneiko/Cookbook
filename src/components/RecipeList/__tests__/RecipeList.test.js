/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import RecipeList from '../RecipeList';
import Recipe from '../../Recipe/Recipe';

describe('RecipeList component', () => {
  const props = {
    items: [{ id: 1 }],
    onDeleteRecipe: () => null,
    onEditRecipe: () => null,
  };

  const newRecipeList = shallow(<RecipeList {...props} />);

  it('renders a properly', () => {
    expect(newRecipeList).toMatchSnapshot();
  });

  it('renders one ul', () => {
    expect(newRecipeList.find('ul.recipes__list')).toHaveLength(1);
  });

  it('renders three li', () => {
    expect(newRecipeList.find('li.recipes__item')).toHaveLength(1);
  });

  it('renders a Recipe', () => {
    expect(newRecipeList.find(Recipe)).toHaveLength(1);
  });
});
