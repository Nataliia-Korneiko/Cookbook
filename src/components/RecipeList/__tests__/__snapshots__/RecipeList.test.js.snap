// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList component renders a properly 1`] = `ShallowWrapper {}`;
